lib/web_ui/test/golden_tests/engine/text_placeholders_test.dart

```diff
@@ -39,6 +40,7 @@ void testMain() async {
       );
       offset = offset.translate(0.0, 80.0);
     }
+    recordingCanvas.restore();
```

Comment:
i just see save/restore pairs. why was it leaking when restore is not called ?

Reply:
Because of the way canvas pool works lazily now? I saw you added `restore` calls in multiple tests: https://github.com/flutter/engine/pull/15153